data/tabular/check_smiles_split.py

```diff
@@ -0,0 +1,200 @@
+"""This script checks for data leakage in the splits of a tabular dataset."""
+import os
+from glob import glob
+from pathlib import Path
+from typing import List, Literal, Union
+
+import dask.dataframe as dd
+import fire
+import yaml
+
+
+def get_all_yamls(data_dir):
+    """Returns all yaml files in the data directory"""
+    return glob(str(Path(data_dir) / "**" / "*.yaml"), recursive=True)
```

Comment:
Should this be run from `data/{tabular,kg}` or from `data` to check the `tabular` and `kg` dirs?